context.go

```diff
@@ -157,21 +176,37 @@ func merge[K comparable, V any](a, b map[K][]V) (merged map[K][]V) {
 	return
 }
 
+// encodeOneAddressType encodes the given addressData values and returns the corresponding WAF object and its refs.
+// If the addressData is empty, it returns nil for the WAF object and an empty ref pool.
+// At this point, the only error we can get is an error in case the top level object is a nil map, but this
+// behaviour is expected since either persistent or ephemeral addresses are allowed to be null one at a time.
+// In this case, EncodeAddresses will return nil contrary to Encode which will return an nil wafObject,
+// which is what we need to send to ddwaf_run to signal that the address data is empty.
+func (context *Context) encodeOneAddressType(addressData map[string]any, runTimings *Timings) (*bindings.WafObject, *encoder, error) {
+	var data *bindings.WafObject = nil
+	encoder := newLimitedEncoder(runTimings.Remaining())
```

Comment:
This appears to extract the `Remaining` from the `runTimings`, which means the encoder might be operating based on a stale deadline?